### ISSUE TITLE
Mocha-aware prefer-arrow-callback

### ISSUE THREAD

**Issue by paulmelnikow** [lo1tuma/eslint-plugin-mocha#141]
As discussed at eslint/eslint#9255, it would be great to have a Mocha-aware version of `prefer-arrow-callback` which ignored Mocha globals but triggered errors for everything else.

I would turn this on in a jiffy!

**Comment by vhmth** [comment]
Could anyone point me in the proper direction for which files to generally look at to implement this? We would definitely love to have this at Loom. Currently we need two separate ESLint configs (one for our tests and one for all other stuff).

**Comment by paulmelnikow** [comment]
I wrote the [no-mocha-arrows](https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/lib/rules/no-mocha-arrows.js) rule. You need to add a file in `lib/rules`, another in `test/rules`, and also update the docs. I'd suggest reading that one, and also the [eslint core rule for prefer-arrow-callback](https://github.com/eslint/eslint/blob/master/lib/rules/prefer-arrow-callback.js).

**Comment by astorije** [comment]
In terms of usage, would making it similar to [`eslint-plugin-chai-friendly`](https://github.com/ihordiachenko/eslint-plugin-chai-friendly#usage)'s `no-unused-expressions` make sense?
I.e. in a test-specific `.eslintrc.yml`, one would set it up with:

```yaml
rules:
  prefer-arrow-callback: off
  mocha/prefer-arrow-callback: error
```

I too would happily use it and [get rid of my ugly `eslint-disable-line` comments](https://github.com/astorije/chai-immutable/blob/78fa174428709bd786fba90b9693e9dcbd11d101/test/test.js#L58-L60) :)

**Comment by basickarl** [comment]
Any update on this? Adding two files isn't the way to go fix this. this should be automatically apart of this library.

**Comment by lo1tuma** [comment]
Thanks to @quarklemotion this has been shipped in version `5.1.0`.
